includes/fbbot/class-omise-fbbot-payment-handler.php

```diff
@@ -0,0 +1,253 @@
+    $this->public_key  = $this->test_mode ? $settings['test_public_key'] : $settings['live_public_key'];
+
+    if ( empty( $this->private_key ) || empty( $this->public_key ) ) {
+      return;
+    }
+
+    // Set omise key
+    $this->set_oms_key();
+
+    function load_scripts() {
+      $settings = get_option( 'woocommerce_omise_settings', null );
+      $test_mode = isset( $settings['sandbox'] ) && $settings['sandbox'] == 'yes';
+      $public_key = $test_mode ? $settings[ "test_public_key" ] : $settings[ "live_public_key" ];
+
+      // Note : Set for testing on staging
+      // wp_enqueue_script( 'omise-js', 'https://omise-cdn.s3.amazonaws.com/assets/frontend-libs/staging-omise.js', array( 'jquery' ), OMISE_WOOCOMMERCE_PLUGIN_VERSION, true );
+
+      wp_enqueue_script( 'omise-js', 'https://cdn.omise.co/omise.js', array( 'jquery' ), OMISE_WOOCOMMERCE_PLUGIN_VERSION, true );
+    
+      wp_enqueue_script( 'omise-util', plugins_url( 'omise-woocommerce/assets/javascripts/omise-util.js' ), array( 'omise-js' ), OMISE_WOOCOMMERCE_PLUGIN_VERSION, true );
+
+      wp_enqueue_script( 'omise-payment-on-messenger-form-handler', plugins_url( 'omise-woocommerce/assets/javascripts/omise-payment-on-messenger-form-handler.js'), array( 'jquery' ), OMISE_WOOCOMMERCE_PLUGIN_VERSION, true);
+
+      wp_localize_script( 'omise-payment-on-messenger-form-handler', 'omise_params', array(
+        'key'       => $public_key,
+        'vault_url' => OMISE_VAULT_HOST
+      ) );
+    }
+
+    add_action( 'wp_enqueue_scripts', 'load_scripts' );
+	}
+
+  function set_oms_key() {
+    define( 'OMISE_PUBLIC_KEY', $this->public_key );
+    define( 'OMISE_SECRET_KEY', $this->private_key );
+  }
+
+	public static function get_instance() {
+		if ( ! self::$instance ) {
+			self::$instance = new self();
+		}
+
+		return self::$instance;
+	}
+
+	public function parameter_queryvars( $qvars ) {
+		$qvars[] = 'product_id';
+		$qvars[] = 'messenger_id';
+    $qvars[] = 'error_message';
+
+		return $qvars;
+	}
+
+	public function payment_page_detect( $posts ) {
+		global $wp;
+    global $wp_query;
+		global $payment_page_detect;
+    
+    $payment_page_url = "pay-on-messenger";
+    $payment_purchase_complete = "complete-payment";
+    $payment_error_url = "pay-on-messenger-error";
+
+    if ( strtolower( $wp->request ) == $payment_purchase_complete ) {
+      $image_url = site_url() . '/wp-content/plugins/omise-woocommerce/assets/images/omise_logo.png';
+      $url = 'https://www.messenger.com/closeWindow/?image_url=' . $image_url . '&display_text=THANKS%20FOR%20PURCHASE';
+      if ( wp_redirect( $url ) ) {
+        exit;
+      }
+
+      return $posts;
+    }
+
+    // Create custom page
+    $post = new stdClass;
+
+    if ( strtolower( $wp->request ) == $payment_page_url ) {
+      $post->post_title = __( 'Your order' );
+      $post->post_content = $this->payment_page_render();
+    }
+
+    if ( strtolower( $wp->request ) == $payment_error_url ) {
```

Comment:
@igroomgrim how about consider to have a delegated method to work on this thing?

something like, 
```php
if ( $this->is_omise_payment_page() ) {
    $post = $this->build_payment_page();
} else if ( $this->is_omise_payment_error_page() ) {
    $post = $this->build_payment_error_page();
}

if ( $this->is_accessible() ) {
    $posts = array( $post );

    // make wp_query
    $wp_query->is_page             = true;
    $wp_query->is_singular         = true;
    $wp_query->is_home             = false;
    $wp_query->is_archive          = false;
    $wp_query->is_category         = false;
    $wp_query->is_404              = false;
    $wp_query->query_vars['error'] = '';
}

return $posts;
```